src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PreferIsExactSpellingIsTrueForKnownApisAnalyzer.cs

```diff
@@ -0,0 +1,155 @@
+                                                                             isDataflowRule: false,
+                                                                             isEnabledByDefaultInFxCopAnalyzers: true);
+
+        internal static DiagnosticDescriptor WideRule = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessageWideSuffix,
+                                                                             DiagnosticCategory.Performance,
+                                                                             RuleLevel.IdeSuggestion,
+                                                                             description: s_localizableDescription,
+                                                                             isPortedFxCopRule: false,
+                                                                             isDataflowRule: false,
+                                                                             isEnabledByDefaultInFxCopAnalyzers: true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(DefaultRule, WideRule);
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            analysisContext.RegisterSymbolAction(AnalyzeExternMethod, SymbolKind.Method);
+        }
+
+        private static void AnalyzeExternMethod(SymbolAnalysisContext context)
+        {
+            var methodSymbol = (IMethodSymbol)context.Symbol;
+            if (!methodSymbol.IsExtern)
+            {
+                return;
+            }
+
+            if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeInteropServicesDllImportAttribute, out var dllImportType))
+            {
+                return;
+            }
+
+            var dllImportAttribute = methodSymbol
+                .GetAttributes()
+                .FirstOrDefault(x => x.AttributeClass.Equals(dllImportType));
+
+            if (dllImportAttribute is null)
+            {
+                return;
+            }
+
+            var hasExactSpellingParameter = dllImportAttribute.NamedArguments.FirstOrDefault(x => x.Key.Equals("ExactSpelling", StringComparison.Ordinal));
+            var hasCharSetParameter = dllImportAttribute.NamedArguments.FirstOrDefault(x => x.Key.Equals("CharSet", StringComparison.Ordinal));
+
+            var methodName = methodSymbol.GetActualExternName(dllImportAttribute);
+            var isExactSpelling = hasExactSpellingParameter.Key is not null && bool.TryParse(hasExactSpellingParameter.Value.Value.ToString(), out var isExact) && isExact;
```

Comment:
Seems like the second `.Value` access can throw: http://sourceroslyn.io/#Microsoft.CodeAnalysis/Symbols/TypedConstant.cs,99. You need a Kind check. Additionally, also seems the returned value can be `null`, so you need a `?.ToString()`